tests/text/helpers.py

```diff
@@ -0,0 +1,538 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pickle
+import sys
+from enum import Enum, unique
+from functools import partial
+from typing import Any, Callable, Dict, Sequence, Union
+
+import numpy as np
+import pytest
+import torch
+from torch import Tensor
+from torch.multiprocessing import Pool, set_start_method
+
+from tests.helpers.testers import NUM_PROCESSES, setup_ddp
+from torchmetrics import Metric
+
+try:
+    set_start_method("spawn")
+except RuntimeError:
+    pass
+
+
+@unique
+class INPUT_ORDER(Enum):
+    PREDS_FIRST = 1
+    TARGETS_FIRST = 2
+
+
+TEXT_METRIC_INPUT = Union[Sequence[str], Sequence[Sequence[str]], Sequence[Sequence[Sequence[str]]]]
+NUM_BATCHES = 2
+
+
+def _assert_allclose(pl_result: Any, sk_result: Any, atol: float = 1e-8, key: str = None):
+    """Utility function for recursively asserting that two results are within a certain tolerance."""
+    # single output compare
+    if isinstance(pl_result, Tensor):
+        assert np.allclose(pl_result.detach().cpu().numpy(), sk_result, atol=atol, equal_nan=True)
+    # multi output compare
+    elif isinstance(pl_result, Sequence):
+        for pl_res, sk_res in zip(pl_result, sk_result):
+            _assert_allclose(pl_res, sk_res, atol=atol)
+    elif isinstance(pl_result, Dict):
+        if key is None:
+            raise KeyError("Provide Key for Dict based metric results.")
+        assert np.allclose(pl_result[key].detach().cpu().numpy(), sk_result, atol=atol, equal_nan=True)
```

Comment:
lets just add this part (and the `key` argument) to the original `_assert_allclose` and then you can just import it here